src/Microsoft.OData.Client/DataServiceQueryOfT.cs

```diff
@@ -314,14 +314,27 @@ public DataServiceQuery<TElement> Expand<TTarget>(Expression<Func<TElement, TTar
         /// <returns>A new <see cref="T:Microsoft.OData.Client.DataServiceQuery`1" /> object that has the inline count option set.</returns>
         public DataServiceQuery<TElement> IncludeTotalCount()
         {
-            MethodInfo mi = typeof(DataServiceQuery<TElement>).GetMethod("IncludeTotalCount");
+            MethodInfo mi = typeof(DataServiceQuery<TElement>).GetMethods()
+                .First(m => m.Name == "IncludeTotalCount" && m.GetParameters().Length == 0);
 
             return (DataServiceQuery<TElement>)this.Provider.CreateQuery<TElement>(
                 Expression.Call(
                     Expression.Convert(this.Expression, typeof(DataServiceQuery<TElement>.DataServiceOrderedQuery)),
                     mi));
         }
 
+        /// <summary>Requests that the count of all entities in the entity set be returned inline with the query results.</summary>
+        /// <returns>A new <see cref="T:Microsoft.OData.Client.DataServiceQuery`1" /> object that has the inline count option set.</returns>
+        /// <param name="includeTotal">Whether to include total count.</param>
+        public DataServiceQuery<TElement> IncludeTotalCount(bool includeTotal)

```

Comment:
The 
C# provides an optional arguments feature which may have some candidate use here
The signature could be
```public DataServiceQuery<TElement> IncludeTotalCount(bool? includeTotal = false)```
[More info](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/named-and-optional-arguments#optional-arguments)
However considering that you are doing some reflection stuff to attach the parameterless method above to the expression, maybe old school overloading is how to do it...

Reply:
I have updated the code to ensure we add `$count=false` to the request Uri whenever we call `IncludeTotal(false)`